src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -167,11 +370,152 @@ public int getBonusVision()
         return MAX_VISION_BONUS_MULTIPLIER * VISION_BONUS;
     }
 
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        task = Task.values()[compound.getInteger(TAG_TASK)];
+        int jobId = compound.getInteger(TAG_JOB);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'jobId' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)